src/electionguard-ui/ElectionGuard.UI/Helpers/ElectionRecordGenerator.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.IO.Compression;
+using System.Text.Json;
+using CommunityToolkit.Mvvm.DependencyInjection;
+using ElectionGuard.ElectionSetup;
+using ElectionGuard.UI.Lib.Services;
+using MongoDB.Driver;
+
+namespace ElectionGuard.UI.Helpers;
+
+public static class ElectionRecordGenerator
+{
+    private static readonly string GUARDIAN_FOLDER = "guardians";
+    private static readonly string DEVICE_FOLDER = "encryption_devices";
+    private static readonly string BALLOT_FOLDER = "submitted_ballots";
+
+    private static readonly string GUARDIAN_PREFIX = "guardian_";
+    private static readonly string DEVICE_PREFIX = "device_";
+
+
+    public static async Task GenerateEelectionRecord(TallyRecord tally, string outputFolder)
+    {
+        // check the state of the tally
+        if (tally.State != TallyState.Complete)
+        {
+            throw new ArgumentException("The tally must be completed before generating an election record.");
+        }
+
+        // generate temp path to export all of the files to
+        var tempFolder = Directory.CreateTempSubdirectory("eg_");
+
+        // export the guardians
+        await ExportGuardiansAsync(Path.Combine(tempFolder.FullName, GUARDIAN_FOLDER), tally.KeyCeremonyId);
+
+        // export the devices
+        await ExportDevices(Path.Combine(tempFolder.FullName, DEVICE_FOLDER), tally.ElectionId);

```

Comment:
Guardians and Devices might not need to be exported for v2, but for v1 it is exported.  These 2 methods could be removed or a setting on the generation to give compatibility for both version.